packages/webapp/pages/join.tsx

```diff
@@ -1,71 +1,41 @@
-import {
-  downloadBrowserExtension,
-  isDevelopment,
-} from '@dailydotdev/shared/src/lib/constants';
-import { PublicProfile } from '@dailydotdev/shared/src/lib/user';
+import { isDevelopment } from '@dailydotdev/shared/src/lib/constants';
 import request from 'graphql-request';
-import React, { ReactElement, useContext, useEffect, useState } from 'react';
-import BrowsersIcon from '@dailydotdev/shared/icons/browsers.svg';
-import { Image } from '@dailydotdev/shared/src/components/image/Image';
-import { cloudinary } from '@dailydotdev/shared/src/lib/image';
-import PlayIcon from '@dailydotdev/shared/src/components/icons/Play';
-import { IconSize } from '@dailydotdev/shared/src/components/Icon';
-import CloseButton from '@dailydotdev/shared/src/components/CloseButton';
-import {
-  ModalKind,
-  ModalSize,
-} from '@dailydotdev/shared/src/components/modals/common/types';
-import { Modal } from '@dailydotdev/shared/src/components/modals/common/Modal';
-import { useRouter } from 'next/router';
-import AuthContext from '@dailydotdev/shared/src/contexts/AuthContext';
-import useMedia from '@dailydotdev/shared/src/hooks/useMedia';
-import { laptop } from '@dailydotdev/shared/src/styles/media';
+import React, { FunctionComponent, ReactElement, useEffect } from 'react';
 import { GetServerSideProps } from 'next';
 import { NextSeo, NextSeoProps } from 'next-seo';
-import { useAnalyticsContext } from '@dailydotdev/shared/src/contexts/AnalyticsContext';
 import { setCookie } from '@dailydotdev/shared/src/lib/cookie';
 import { oneYear } from '@dailydotdev/shared/src/lib/dateFormat';
+import { useReferralConfig } from '@dailydotdev/shared/src/hooks/referral/useReferralConfig';
 import { defaultOpenGraph } from '../next-seo';
+import { JoinPageProps } from '../components/invite/common';
+import { AISearchInvite } from '../components/invite/AISearchInvite';
+import Custom404Seo from './404';
 
-type PageProps = {
-  referringUser: Pick<PublicProfile, 'id' | 'name' | 'image'>;
-  campaign: ReferralCampaignKey;
-};
+type ReferralRecord<T> = Record<ReferralCampaignKey, T>;
 
-const Page = ({ referringUser, campaign }: PageProps): ReactElement => {
-  const router = useRouter();
-  const { trackEvent } = useAnalyticsContext();
-  const { user } = useContext(AuthContext);
-  const [isVideoOpen, setVideoOpen] = useState(false);
-  const isDesktop = useMedia([laptop.replace('@media ', '')], [true], false);
+const componentsMap: ReferralRecord<FunctionComponent<JoinPageProps>> = {
+  search: AISearchInvite,
+};
 
-  const onVideoModalClose = () => {
-    setVideoOpen(false);
-  };
+const Page = ({
+  referringUser,
+  campaign,
+  token,
+}: JoinPageProps): ReactElement => {
+  const Component = componentsMap[campaign];
+  const { title, description, images, redirectTo } = useReferralConfig({
+    campaign,
+    referringUser,
+  });
 
   useEffect(() => {
-    if (!user?.id) {
+    if (!Object.hasOwn(componentsMap, campaign)) {
```

Comment:
Why not just `componentsMap[campaign]`? ..its more intuitive and has to check less thing internally.

Reply:
Don't wanna add more dependency in the array 😅

Reply:
Since component map is defined outside of component and it is not reactive you will not have to add it to array either way.

Reply:
Right. Thought you wanted to use the already defined variable `const Component = componentsMap[campaign]`. In that case, we can definitely switch to that.